src/main/java/org/mifos/connector/bulk/camel/routes/BatchDetailRoute.java

```diff
@@ -0,0 +1,180 @@
+package org.mifos.connector.bulk.camel.routes;
+
+import org.apache.camel.Exchange;
+import org.apache.camel.LoggingLevel;
+import org.apache.camel.model.dataformat.JsonLibrary;
+import org.mifos.connector.bulk.schema.*;
+import org.mifos.connector.bulk.utils.Utils;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.mifos.connector.bulk.zeebe.ZeebeVariables.*;
+
+@Component
+public class BatchDetailRoute extends BaseRouteBuilder {
+
+    @Value("${config.completion-threshold-check.completion-threshold}")
+    private int completionThreshold;
+
+    private static final String OPS_APP_ACCESS_TOKEN = "opsAppAccessToken";
+
+    @Override
+    public void configure() throws Exception {
+
+        from(RouteId.BATCH_DETAIL.getValue())
+                .id(RouteId.BATCH_DETAIL.getValue())
+                .log("Starting route " + RouteId.BATCH_DETAIL.name())
+//                .to("direct:get-access-token")
+//                .choice()
+//                .when(exchange -> exchange.getProperty(OPS_APP_ACCESS_TOKEN, String.class) != null)
+//                .log(LoggingLevel.INFO, "Got access token, moving on to API call")
+                .to("direct:batch-detail")
+                .to("direct:batch-detail-response-handler");
+//                .otherwise()
+//                .log(LoggingLevel.INFO, "Authentication failed.")
+//                .endChoice();
+
+        getBaseExternalApiRequestRouteDefinition("batch-detail", HttpRequestMethod.GET)
+                .setHeader(
+                        Exchange.REST_HTTP_QUERY,
+                        simple(
+                                BATCH_ID + "=${exchangeProperty." + BATCH_ID + "}&" +
+                                        PAGE_NO + "=${exchangeProperty." + PAGE_NO + "}&" +
+                                        PAGE_SIZE + "=${exchangeProperty." + PAGE_SIZE + "}"
+                        )
+                )
+//                .setHeader("Authorization", simple("Bearer ${exchangeProperty."+OPS_APP_ACCESS_TOKEN+"}"))
+//                .setHeader("Platform-TenantId", simple("${exchangeProperty." + TENANT_ID + "}"))
+                .setHeader("Platform-TenantId", simple("rhino"))
+                .process(exchange -> {
+                    logger.info(exchange.getIn().getHeaders().toString());
+                })
+//                .toD(operationsAppConfig.batchDetailUrl +  "?bridgeEndpoint=true&throwExceptionOnFailure=false")
+                .toD("http://localhost:8080/mockapi/v1/batch/detail" +  "?bridgeEndpoint=true&throwExceptionOnFailure=false")
```

Comment:
Is the mock server part of this service itself? otherwise the host URL should be taken from a configuration. Hard coding is not allowed